### PR TITLE
Entity manger remove method

### PR DESCRIPTION
**Pull request by yuriyfomin17**
- added remove method implementation
- added test form remove method
- added util method for finding table name, id value and id field name

### CODE REVIEW

**Review by artemyankovets — APPROVED** [review]

**Comment by syevtushok** [comment]
CleanShot 2023-03-27 at 10 31 50@2x
your tests don't work for me. And I thought we use H2 for tests?

File: pom.xml [hunk 1]
```diff
@@ -39,6 +39,17 @@
...
         </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <version>5.9.2</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.postgresql</groupId>
+            <artifactId>postgresql</artifactId>
```
Comment (Mikhail-Mikhailenko) [inline pom.xml]: We should not use driver for our library. We should use H2 in memory DB. We already have environtment for tests. 
image

File: src/main/java/com/breskul/bibernate/persistence/util/DaoUtils.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+public class DaoUtils {
+
+    public static Field getIdentifierField(Class<?> entityClass){
+        var cause = "entity is not marked with Id annotation";
+        var solution = "mark id column with Id annotation";
+        return Arrays.stream(entityClass.getDeclaredFields())
+                .filter(field -> field.isAnnotationPresent(Id.class))
+                .findAny()
+                .orElseThrow(() -> new DaoUtilsException(cause, solution));
+    }
```
Comment (Mikhail-Mikhailenko) [inline src/main/java/com/breskul/bibernate/persistence/util/DaoUtils.java:20]: miss spacing

File: src/test/java/com/breskul/bibernate/persistence/EntityManagerImplTest.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    private static DataSource dataSource;
+
+    @BeforeAll
+    public static void initDataSource() {
+        PGSimpleDataSource pgSimpleDataSource = new PGSimpleDataSource();
+        pgSimpleDataSource.setURL(DB_URL);
+        dataSource = pgSimpleDataSource;
+        doInConnection(connection -> {
+            try {
+                PreparedStatement createTable = connection.prepareStatement(CREATE_TABLE);
```
Comment (Mikhail-Mikhailenko) [inline src/test/java/com/breskul/bibernate/persistence/EntityManagerImplTest.java]: use init file for creation tables and executing queries for your tests.
And I mentioned above about already created test environtment. 
image
Init.sql will be execute before your tests

**Comment by yuriyfomin17** [comment]
> CleanShot 2023-03-27 at 10 31 50@2x
> 
> your tests don't work for me. And I thought we use H2 for tests?

they do work if you will install Postgres. We did not specify in meetings that for tests we will use H2. Correct  me if I am wrong

**Comment by Mikhail-Mikhailenko** [comment]
> > CleanShot 2023-03-27 at 10 31 50@2x
> > your tests don't work for me. And I thought we use H2 for tests?
> 
> they do work if you will install Postgres. We did not specify in meetings that for tests we will use H2. Correct me if I am wrong

@yuriyfomin17 We don't use real connection for our test. They should work independent from real DB server

File: pom.xml (hunk 1 above)
Comment (yuriyfomin17) [reply pom.xml]: fixed

File: src/test/java/com/breskul/bibernate/persistence/EntityManagerImplTest.java (hunk 3 above)
Comment (yuriyfomin17) [reply src/test/java/com/breskul/bibernate/persistence/EntityManagerImplTest.java]: fixed

**Review by syevtushok — APPROVED** [review]

File: src/test/java/com/breskul/bibernate/persistence/EntityManagerImplTest.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import org.junit.jupiter.api.*;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.function.Consumer;
+
+@TestClassOrder(ClassOrderer.OrderAnnotation.class)
+public class EntityManagerImplTest {
```
Comment (Mikhail-Mikhailenko) [inline src/test/java/com/breskul/bibernate/persistence/EntityManagerImplTest.java:17]: You can extends from AbstractDataSourceTest and use already created data source 
image

**Pull request merged**